src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionCartForm.php

```diff
@@ -0,0 +1,165 @@
+ */
+class PunchoutCatalogConnectionCartForm extends AbstractType
+{
+    protected const FIELD_LABEL_MAX_DESCRIPTION_LENGTH = 'Set Description length on "Transfer to Requisition"';
+    protected const FIELD_LABEL_ENCODING = 'Cart Encoding';
+    protected const FIELD_LABEL_MAPPING = 'Cart Mapping';
+    protected const FIELD_LABEL_DEFAULT_SUPPLIER_ID = 'Default Supplier ID';
+
+    protected const MAX_DESCRIPTION_LENGTH = 99999;
+
+    protected const TEMPLATE_PATH_MAX_DESCRIPTION_LENGTH_FIELD = '@PunchoutCatalogs/ConnectionForm/max_description_length.twig';
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $this->addMaxDescriptionLengthField($builder)
+            ->addEncodingField($builder)
+            ->addMappingField($builder)
+            ->addMaxDescriptionLengthField($builder)
+            ->addDefaultSupplierIdField($builder);
+    }
+
+    /**
+     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     *
+     * @return void
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => PunchoutCatalogConnectionCartTransfer::class,
+        ]);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addMaxDescriptionLengthField(FormBuilderInterface $builder)
+    {
+        $builder->add(PunchoutCatalogConnectionCartTransfer::MAX_DESCRIPTION_LENGTH, IntegerType::class, [
+            'label' => static::FIELD_LABEL_MAX_DESCRIPTION_LENGTH,
+            'required' => false,
+            'constraints' => [
+                new Range([
+                    'min' => 16,
+                    'max' => static::MAX_DESCRIPTION_LENGTH,
+                ]),
+            ],
+            'attr' => [
+                'template_path' => static::TEMPLATE_PATH_MAX_DESCRIPTION_LENGTH_FIELD,
+            ],
+        ]);
+
+        $builder->get(PunchoutCatalogConnectionCartTransfer::MAX_DESCRIPTION_LENGTH)
+            ->addViewTransformer($this->createMaxDescriptionLengthTransformer());
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addEncodingField(FormBuilderInterface $builder)
+    {
+        $builder->add(PunchoutCatalogConnectionCartTransfer::ENCODING, ChoiceType::class, [
+            'label' => static::FIELD_LABEL_ENCODING,
+            'choices' => [
+                'base64' => 'base64',
+                'url-encoded' => 'url-encoded',
+                'no-encoding' => 'no-encoding',
+            ],
+            'constraints' => new NotBlank(),
```

Comment:
'constraints' represents a collection of constraints, not a single one.
Sure, you can set one constraint, but it's not a good practice.

Please use an array here:
```
'constraints' => [
    new NotBlank(),
];
```